In [45]:
import pandas as pd
import numpy as np
data_school = '/Users/vathanahim/Documents/UNCG-FALL2020/IAF603/IAF603_Assignment2/Chicago_Public_Schools_-_Progress_Report_Cards__2011-2012-v3.csv'
data_census = '/Users/vathanahim/Documents/UNCG-FALL2020/IAF603/IAF603_Assignment2/Census_Data_-_Selected_socioeconomic_indicators_in_Chicago__2008___2012-v2.csv'

# PART 1

In [46]:
df_census = pd.read_csv(data_census)
df_school = pd.read_csv(data_school)
pd.set_option("display.max_columns", 200)
pd.set_option("display.max_rows", 200)

# Census Data

In [47]:
df_census.head(5)

COMMUNITY_AREA_NUMBER COMMUNITY_AREA_NAME  PERCENT OF HOUSING CROWDED  \
0                    1.0         Rogers Park                         7.7   
1                    2.0          West Ridge                         7.8   
2                    3.0              Uptown                         3.8   
3                    4.0      Lincoln Square                         3.4   
4                    5.0        North Center                         0.3   

   PERCENT HOUSEHOLDS BELOW POVERTY  PERCENT AGED 16+ UNEMPLOYED  \
0                              23.6                          8.7   
1                              17.2                          8.8   
2                              24.0                          8.9   
3                              10.9                          8.2   
4                               7.5                          5.2   

   PERCENT AGED 25+ WITHOUT HIGH SCHOOL DIPLOMA  \
0                                          18.2   
1                                          20.8   
2                                          11.8   
3                                          13.4   
4                                           4.5   

   PERCENT AGED UNDER 18 OR OVER 64  PER_CAPITA_INCOME   HARDSHIP_INDEX  
0                              27.5               23939            39.0  
1                              38.5               23040            46.0  
2                              22.2               35787            20.0  
3                              25.5               37524            17.0  
4                              26.2               57123             6.0

In [48]:
pd.set_option("display.max_rows", 100)
df_census.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 78 entries, 0 to 77
Data columns (total 9 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   COMMUNITY_AREA_NUMBER                         77 non-null     float64
 1   COMMUNITY_AREA_NAME                           78 non-null     object 
 2   PERCENT OF HOUSING CROWDED                    78 non-null     float64
 3   PERCENT HOUSEHOLDS BELOW POVERTY              78 non-null     float64
 4   PERCENT AGED 16+ UNEMPLOYED                   78 non-null     float64
 5   PERCENT AGED 25+ WITHOUT HIGH SCHOOL DIPLOMA  78 non-null     float64
 6   PERCENT AGED UNDER 18 OR OVER 64              78 non-null     float64
 7   PER_CAPITA_INCOME                             78 non-null     int64  
 8   HARDSHIP_INDEX                                77 non-null     float64
dtypes: float64(7), int64(1), object(1)
memory usage: 5.6+ KB


In [49]:
print("Dimension: " +str(df_census.shape))
df_census.isnull().sum()

Dimension: (78, 9)


COMMUNITY_AREA_NUMBER                           1
COMMUNITY_AREA_NAME                             0
PERCENT OF HOUSING CROWDED                      0
PERCENT HOUSEHOLDS BELOW POVERTY                0
PERCENT AGED 16+ UNEMPLOYED                     0
PERCENT AGED 25+ WITHOUT HIGH SCHOOL DIPLOMA    0
PERCENT AGED UNDER 18 OR OVER 64                0
PER_CAPITA_INCOME                               0
HARDSHIP_INDEX                                  1
dtype: int64

The df_census dataframe contains census data that shows the socio-economic conditions in Chicago. The three primary data
type in this data frame is float, int, and pandas object. This dataframe has 9 variables (columns) and 
78 observations (rows). When checking for the number of missing values in this dataframe, it was founded that columns
'COMMUNITY_AREA_NUMBER' and 'HARDSHIP_INDEX' have one missing value. 

# School Data

In [50]:
df_school.head(5)

School ID                                     NAME_OF_SCHOOL  \
0     610038                  Abraham Lincoln Elementary School   
1     610281  Adam Clayton Powell Paideia Community Academy ...   
2     610185                Adlai E Stevenson Elementary School   
3     609993                    Agustin Lara Elementary Academy   
4     610513                      Air Force Academy High School   

  Elementary, Middle, or High School          Street Address     City State  \
0                                 ES        615 W Kemper Pl   Chicago    IL   
1                                 ES  7511 S South Shore Dr   Chicago    IL   
2                                 ES     8010 S Kostner Ave   Chicago    IL   
3                                 ES     4619 S Wolcott Ave   Chicago    IL   
4                                 HS        3630 S Wells St   Chicago    IL   

   ZIP Code    Phone Number  \
0     60614  (773) 534-5720   
1     60649  (773) 535-6650   
2     60652  (773) 535-2280   
3     60609  (773) 535-4389   
4     60609  (773) 535-1590   

                                               Link   \
0  http://schoolreports.cps.edu/SchoolProgressRep...   
1  http://schoolreports.cps.edu/SchoolProgressRep...   
2  http://schoolreports.cps.edu/SchoolProgressRep...   
3  http://schoolreports.cps.edu/SchoolProgressRep...   
4  http://schoolreports.cps.edu/SchoolProgressRep...   

                      Network Manager                  Collaborative Name  \
0        Fullerton Elementary Network  NORTH-NORTHWEST SIDE COLLABORATIVE   
1           Skyway Elementary Network            SOUTH SIDE COLLABORATIVE   
2           Midway Elementary Network        SOUTHWEST SIDE COLLABORATIVE   
3         Pershing Elementary Network        SOUTHWEST SIDE COLLABORATIVE   
4  Southwest Side High School Network        SOUTHWEST SIDE COLLABORATIVE   

  Adequate Yearly Progress Made?  Track Schedule  \
0                              No       Standard   
1                              No        Track_E   
2                              No       Standard   
3                              No        Track_E   
4                             NDA       Standard   

  CPS Performance Policy Status CPS Performance Policy Level  \
0              Not on Probation                      Level 1   
1              Not on Probation                      Level 1   
2              Not on Probation                      Level 2   
3              Not on Probation                      Level 1   
4              Not on Probation              Not Enough Data   

  HEALTHY_SCHOOL_CERTIFIED Safety Icon   SAFETY_SCORE Family Involvement Icon  \
0                      Yes  Very Strong          99.0             Very Strong   
1                       No      Average          54.0                  Strong   
2                       No       Strong          61.0                     NDA   
3                       No      Average          56.0                 Average   
4                      Yes      Average          49.0                  Strong   

  Family Involvement Score Environment Icon   Environment Score  \
0                       99            Strong               74.0   
1                       66            Strong               74.0   
2                      NDA           Average               50.0   
3                       44           Average               45.0   
4                       60            Strong               60.0   

  Instruction Icon   Instruction Score Leaders Icon  Leaders Score   \
0            Strong               66.0        Strong             65   
1       Very Strong               84.0        Strong             63   
2              Weak               36.0           NDA            NDA   
3              Weak               37.0        Strong             65   
4           Average               55.0       Average             45   

  Teachers Icon  Teachers Score Parent Engagement Icon   \
0         Strong             70                  Strong   
1    

In [51]:
df_school.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 566 entries, 0 to 565
Data columns (total 78 columns):
 #   Column                                             Non-Null Count  Dtype  
---  ------                                             --------------  -----  
 0   School ID                                          566 non-null    int64  
 1   NAME_OF_SCHOOL                                     566 non-null    object 
 2   Elementary, Middle, or High School                 566 non-null    object 
 3   Street Address                                     566 non-null    object 
 4   City                                               566 non-null    object 
 5   State                                              566 non-null    object 
 6   ZIP Code                                           566 non-null    int64  
 7   Phone Number                                       566 non-null    object 
 8   Link                                               565 non-null    object 
 9   Network Ma

In [52]:
print("Dimension: " +str(df_school.shape))
df_school.isnull().sum()

Dimension: (566, 78)


School ID                                             0
NAME_OF_SCHOOL                                        0
Elementary, Middle, or High School                    0
Street Address                                        0
City                                                  0
State                                                 0
ZIP Code                                              0
Phone Number                                          0
Link                                                  1
Network Manager                                       0
Collaborative Name                                    0
Adequate Yearly Progress Made?                        0
Track Schedule                                        0
CPS Performance Policy Status                         0
CPS Performance Policy Level                          0
HEALTHY_SCHOOL_CERTIFIED                              0
Safety Icon                                           0
SAFETY_SCORE                                    

The df_school dataframe contains data about public school assesment in the Chicago area. This dataframe has float, int 
and pandas object data types. df_school dataframe consists of 78 variables (columns) and 566 observations (rows). This 
dataframe has missing values in columns: "LINK", "SAFETY_SCORE", "ENVIRONMENT SCORE", "INSTRUCTION SCORE", "AVERAGE_STUDENT_ATTENDANCE","ISAT Exceeding Math %", 'ISAT Exceeding Reading %', "ISAT Value Add Math", and "ISAT Value Add Read". This dataset has no variables with respective to time as each observation information is dependend on the School ID.

In [73]:
l = ['School ID','COMMUNITY_AREA_NUMBER','NAME_OF_SCHOOL','SAFETY_SCORE','Environment Score','Instruction Score','Parent Engagement Score',
     'Average Teacher Attendance','COMMUNITY_AREA_NAME','College Enrollment Rate %']
df_school_selected = df_school[l]

In [74]:
df_school_selected.head(10)

School ID  COMMUNITY_AREA_NUMBER  \
0     610038                      7   
1     610281                     43   
2     610185                     70   
3     609993                     61   
4     610513                     34   
5     610212                     14   
6     609720                      5   
7     610342                     24   
8     610524                      5   
9     610209                     14   

                                      NAME_OF_SCHOOL  SAFETY_SCORE  \
0                  Abraham Lincoln Elementary School          99.0   
1  Adam Clayton Powell Paideia Community Academy ...          54.0   
2                Adlai E Stevenson Elementary School          61.0   
3                    Agustin Lara Elementary Academy          56.0   
4                      Air Force Academy High School          49.0   
5                  Albany Park Multicultural Academy          66.0   
6                Albert G Lane Technical High School          88.0   
7            Albert R Sabin Elementary Magnet School          67.0   
8              Alcott High School for the Humanities          70.0   
9                 Alessandro Volta Elementary School          43.0   

   Environment Score  Instruction Score Parent Engagement Score  \
0               74.0               66.0                      56   
1               74.0               84.0                      46   
2               50.0               36.0                      47   
3               45.0               37.0                      53   
4               60.0               55.0                      53   
5               66.0               71.0                      46   
6               62.0               52.0                     NDA   
7               30.0               18.0                     NDA   
8               67.0               51.0                      57   
9               28.0               37.0                      51   

  Average Teacher Attendance COMMUNITY_AREA_NAME College Enrollment Rate %  
0                     96.40%        LINCOLN PARK                       NDA  
1                     95.30%         SOUTH SHORE                       NDA  
2                     94.70%             ASHBURN                       NDA  
3                     95.80%            NEW CITY                       NDA  
4                     96.90%       ARMOUR SQUARE                       NDA  
5                     96.90%         ALBANY PARK                       NDA  
6                     96.20%        NORTH CENTER                      79.8  
7                     95.00%           WEST TOWN                       NDA  
8                     96.90%        NORTH CENTER                       NDA  
9                     95.90%         ALBANY PARK                       NDA

In [75]:
df_school_selected.isnull().sum()

School ID                      0
COMMUNITY_AREA_NUMBER          0
NAME_OF_SCHOOL                 0
SAFETY_SCORE                  53
Environment Score             53
Instruction Score             53
Parent Engagement Score        0
Average Teacher Attendance     0
COMMUNITY_AREA_NAME            0
College Enrollment Rate %      0
dtype: int64

List-wise deletion was used for schools that doesn't have 'Parent Engagement Score'. However, for 
'College Enrollment Rate', we will not be deleting the missing values 
since only high schools have them and pandas operations will be intrinsically skip those missing values. Instead it will be replace with 'nan. 'SAFETY_SCORE','Environment Score','Instruction Score' missing values will be replace with the mean value of the population for each of their respective column because this is not dependent on the level of education (elementary,middle,high).

In [76]:
#drops rows that conatins missing value for parent Engagement Score (list-wise deletion)
df_school_selected = df_school_selected[~df_school_selected['Parent Engagement Score'].isin(['NDA'])]
df_school_selected['Parent Engagement Score'].isnull().sum()

0

In [77]:
#replacing NDA with nan
df_school_selected = df_school_selected.replace('NDA', np.nan)

In [78]:
#replacing nan with mean for 'SAFETY_SCORE','Environment Score','Instruction Score'
mean_list = ['SAFETY_SCORE','Environment Score','Instruction Score']

for i in mean_list:
    mean = int(df_school_selected[i].mean())
    df_school_selected[i] = df_school_selected[i].replace(np.nan, mean)

In [79]:
df_school_selected.isnull().sum()

School ID                       0
COMMUNITY_AREA_NUMBER           0
NAME_OF_SCHOOL                  0
SAFETY_SCORE                    0
Environment Score               0
Instruction Score               0
Parent Engagement Score         0
Average Teacher Attendance      0
COMMUNITY_AREA_NAME             0
College Enrollment Rate %     404
dtype: int64

# Part 2

In [70]:
import sqlite3

In [71]:
assign2db = "assign2.db"
conn = sqlite3.connect(assign2db)
print(conn)

In [72]:
conn = sqlite3.connect(assign2db)
cursor = conn.cursor()
#Drop public school table if it exits
cursor.execute("DROP TABLE IF EXISTS `PUBLIC_SCHOOL`")
print("Table dropped")
conn.close()

Table dropped


Community area number or community area name can serve as foreign key but it's prefered that a non-string data type is foreign key

In [80]:
#create public school table with appropriate fields
conn = sqlite3.connect(assign2db)
cursor = conn.cursor()
try:
    cursor.execute("""
    CREATE TABLE PUBLIC_SCHOOL (
    SCHOOL_ID INTEGER PRIMARY KEY,
    COMMUNITY_AREA_NUMBER INTEGER,
    NAME_OF_SCHOOL TEXT NOT NULL,
    SAFETY_SCORE FLOAT DEFAULT 0,
    Environment_Score FLOAT DEFAULT 0,
    Instruction_Score FLOAT DEFAULT 0,
    Parent_Engagement_Score INT DEFAULT 0,
    Average_Teacher_Attendance FLOAT DEFAULT 0,
    COMMUNITY_AREA_NAME TEXT NOT NULL,
    College_Enrollment_Rate_perc FLOAT DEFAULT 0
    );
    """)
    print("Success Creation")
except Exception as e:
    print(str(e))
    print("Table creation failed")
finally:
    conn.close()

Success Creation


In [ ]:
#convert to list so it can be inserted into table
school_list = df_school_selected.values.tolist()
school_list

In [83]:
#inserting the fields into the table
conn = sqlite3.connect(assign2db)
cursor = conn.cursor()
try:
    cursor.executemany("""
    INSERT INTO PUBLIC_SCHOOL (SCHOOL_ID, COMMUNITY_AREA_NUMBER, NAME_OF_SCHOOL, SAFETY_SCORE, 
    Environment_Score, Instruction_Score, Parent_Engagement_Score, Average_Teacher_Attendance, COMMUNITY_AREA_NAME,
    College_Enrollment_Rate_perc)
    VALUES (?,?,?,?,?,?,?,?,?,?)
    """, school_list)
    conn.commit()
    print("Insert Successfully")
except Exception as e:
    print(str(e))
    print("Insert failed")
finally:
    conn.close()

Insert Successfully


In [107]:
#test to see if value is inserted properly
conn = sqlite3.connect(assign2db)
cursor = conn.cursor()
cursor.execute("SELECT * FROM PUBLIC_SCHOOL")
result = cursor.fetchall()
conn.close
for i in result:
    print(i)

(609679, 19, 'Charles Allen Prosser Career Academy High School', 59.0, 53.0, 51.0, 46, '94.40%', 'BELMONT CRAGIN', 58.0)
(609682, 61, 'Ellen H Richards Career Academy High School', 30.0, 32.0, 19.0, 43, '95.70%', 'NEW CITY', 43.1)
(609693, 23, 'George Westinghouse High School', 74.0, 77.0, 80.0, 54, '96.80%', 'HUMBOLDT PARK', None)
(609694, 62, 'John Hancock College Preparatory High School', 51.0, 44.0, 41.0, 50, '95.90%', 'WEST ELSDON', 45.9)
(609695, 4, 'Roald Amundsen High School', 51.0, 43.0, 42.0, 43, '95.90%', 'LINCOLN SQUARE', 49.1)
(609698, 70, 'William J Bogan High School', 20.0, 18.0, 29.0, 41, '94.30%', 'ASHBURN', 50.0)
(609702, 28, 'Richard T Crane Technical Preparatory High School', 43.0, 50.0, 48.0, 47, '93.90%', 'NEAR WEST SIDE', 32.6)
(609711, 67, 'William Rainey Harper High School', 22.0, 39.0, 42.0, 49, '94.70%', 'WEST ENGLEWOOD', 40.3)
(609719, 6, 'Lake View High School', 64.0, 52.0, 43.0, 49, '96.20%', 'LAKE VIEW', 62.8)
(609726, 49, 'Gwendolyn Brooks College Prepar

In [90]:
df_census.isnull().sum()

COMMUNITY_AREA_NUMBER                           1
COMMUNITY_AREA_NAME                             0
PERCENT OF HOUSING CROWDED                      0
PERCENT HOUSEHOLDS BELOW POVERTY                0
PERCENT AGED 16+ UNEMPLOYED                     0
PERCENT AGED 25+ WITHOUT HIGH SCHOOL DIPLOMA    0
PERCENT AGED UNDER 18 OR OVER 64                0
PER_CAPITA_INCOME                               0
HARDSHIP_INDEX                                  1
dtype: int64

In [115]:
conn = sqlite3.connect(assign2db)
cursor = conn.cursor()
#Drop public school table if it exits
cursor.execute("DROP TABLE IF EXISTS `CENSUS`")
print("Table dropped")
conn.close()

Table dropped


In [116]:
conn = sqlite3.connect(assign2db)
cursor = conn.cursor()
try:
    cursor.execute("""
    CREATE TABLE CENSUS (
    COMMUNITY_AREA_NUMBER INTEGER DEFAULT 0,
    COMMUNITY_AREA_NAME TEXT NOT NULL,
    PERCENT_OF_HOUSING_CROWDED FLOAT DEFAULT 0,
    PERCENT_HOUSEHOLDS_BELOW_POVERTY FLOAT DEFAULT 0,
    PERCENT_AGED_16_UNEMPLOYED FLOAT DEFAULT 0,
    PERCENT_AGED_25_WITHOUT_HIGH_SCHOOL_DIPLOMA FLOAT DEFAULT 0,
    PERCENT_AGED_UNDER_18_OR_OVER_64 FLOAT DEFAULT 0,
    PER_CAPITA_INCOME INT DEFAULT 0,
    HARDSHIP_INDEX FLOAT DEFAULT 0
    );
    """)
    print("Success Creation")
except Exception as e:
    print(str(e))
    print("Table creation failed")
finally:
    conn.close()

Success Creation


In [117]:
census_list = df_census.values.tolist()
census_list

[[1.0, 'Rogers Park', 7.7, 23.6, 8.7, 18.2, 27.5, 23939, 39.0],
 [2.0, 'West Ridge', 7.8, 17.2, 8.8, 20.8, 38.5, 23040, 46.0],
 [3.0, 'Uptown', 3.8, 24.0, 8.9, 11.8, 22.2, 35787, 20.0],
 [4.0, 'Lincoln Square', 3.4, 10.9, 8.2, 13.4, 25.5, 37524, 17.0],
 [5.0, 'North Center', 0.3, 7.5, 5.2, 4.5, 26.2, 57123, 6.0],
 [6.0, 'Lake View', 1.1, 11.4, 4.7, 2.6, 17.0, 60058, 5.0],
 [7.0, 'Lincoln Park', 0.8, 12.3, 5.1, 3.6, 21.5, 71551, 2.0],
 [8.0, 'Near North Side', 1.9, 12.9, 7.0, 2.5, 22.6, 88669, 1.0],
 [9.0, 'Edison Park', 1.1, 3.3, 6.5, 7.4, 35.3, 40959, 8.0],
 [10.0, 'Norwood Park', 2.0, 5.4, 9.0, 11.5, 39.5, 32875, 21.0],
 [11.0, 'Jefferson Park', 2.7, 8.6, 12.4, 13.4, 35.5, 27751, 25.0],
 [12.0, 'Forest Glen', 1.1, 7.5, 6.8, 4.9, 40.5, 44164, 11.0],
 [13.0, 'North Park', 3.9, 13.2, 9.9, 14.4, 39.0, 26576, 33.0],
 [14.0, 'Albany Park', 11.3, 19.2, 10.0, 32.9, 32.0, 21323, 53.0],
 [15.0, 'Portage Park', 4.1, 11.6, 12.6, 19.3, 34.0, 24336, 35.0],
 [16.0, 'Irving Park', 6.3, 13.1, 10.0, 2

In [118]:
#inserting the fields into the table
conn = sqlite3.connect(assign2db)
cursor = conn.cursor()
try:
    cursor.executemany("""
    INSERT INTO CENSUS (COMMUNITY_AREA_NUMBER,
    COMMUNITY_AREA_NAME,
    PERCENT_OF_HOUSING_CROWDED,
    PERCENT_HOUSEHOLDS_BELOW_POVERTY,
    PERCENT_AGED_16_UNEMPLOYED,
    PERCENT_AGED_25_WITHOUT_HIGH_SCHOOL_DIPLOMA,
    PERCENT_AGED_UNDER_18_OR_OVER_64,
    PER_CAPITA_INCOME,
    HARDSHIP_INDEX)
    VALUES (?,?,?,?,?,?,?,?,?)
    """, census_list)
    conn.commit()
    print("Insert Successfully")
except Exception as e:
    print(str(e))
    print("Insert failed")
finally:
    conn.close()

Insert Successfully


In [119]:
conn = sqlite3.connect(assign2db)
cursor = conn.cursor()
cursor.execute("SELECT * FROM CENSUS")
result = cursor.fetchall()
conn.close()

In [120]:
for i in result:
    print(i)

(1, 'Rogers Park', 7.7, 23.6, 8.7, 18.2, 27.5, 23939, 39.0)
(2, 'West Ridge', 7.8, 17.2, 8.8, 20.8, 38.5, 23040, 46.0)
(3, 'Uptown', 3.8, 24.0, 8.9, 11.8, 22.2, 35787, 20.0)
(4, 'Lincoln Square', 3.4, 10.9, 8.2, 13.4, 25.5, 37524, 17.0)
(5, 'North Center', 0.3, 7.5, 5.2, 4.5, 26.2, 57123, 6.0)
(6, 'Lake View', 1.1, 11.4, 4.7, 2.6, 17.0, 60058, 5.0)
(7, 'Lincoln Park', 0.8, 12.3, 5.1, 3.6, 21.5, 71551, 2.0)
(8, 'Near North Side', 1.9, 12.9, 7.0, 2.5, 22.6, 88669, 1.0)
(9, 'Edison Park', 1.1, 3.3, 6.5, 7.4, 35.3, 40959, 8.0)
(10, 'Norwood Park', 2.0, 5.4, 9.0, 11.5, 39.5, 32875, 21.0)
(11, 'Jefferson Park', 2.7, 8.6, 12.4, 13.4, 35.5, 27751, 25.0)
(12, 'Forest Glen', 1.1, 7.5, 6.8, 4.9, 40.5, 44164, 11.0)
(13, 'North Park', 3.9, 13.2, 9.9, 14.4, 39.0, 26576, 33.0)
(14, 'Albany Park', 11.3, 19.2, 10.0, 32.9, 32.0, 21323, 53.0)
(15, 'Portage Park', 4.1, 11.6, 12.6, 19.3, 34.0, 24336, 35.0)
(16, 'Irving Park', 6.3, 13.1, 10.0, 22.4, 31.6, 27249, 34.0)
(17, 'Dunning', 5.2, 10.6, 10.0, 16.2, 

In [121]:
conn = sqlite3.connect(assign2db)
cursor = conn.cursor()
cursor.execute("SELECT COUNT(*) FROM PUBLIC_SCHOOL")
public_school_count = cursor.fetchone()
cursor.execute("SELECT COUNT(*) FROM CENSUS")
census_count = cursor.fetchone()
conn.close()
print("Total Rows In Public_School Table: " + str(public_school_count[0]))
print("Ttoal Rows In Census Table: " + str(census_count[0]))

Total Rows In Public_School Table: 432
Ttoal Rows In Census Table: 78


The count of rows for public school is 432.
The count of rows for census is 78.